src/Hazelcast.Net/DistributedObjects/IHList.cs

```diff
@@ -34,28 +41,29 @@ public interface IHList<T> : IHCollection<T>
 
         /// <summary>
         /// Inserts all of the elements in the specified collection into this
-        /// list at the specified position (optional operation).  Shifts the
-        /// element currently at that position (if any) and any subsequent
+        /// list at the specified position.
+        /// </summary>
+        /// <param name="index">index at which to insert the first element from the specified collection</param>
+        /// <param name="items">collection containing elements to be added to this list</param>
+        /// <typeparam name="TItem"></typeparam>
+        /// <returns><c>true</c> if this list changed as a result of the call</returns>
+        /// <remarks>
+        /// Shifts the element currently at that position (if any) and any subsequent
         /// elements to the right (increases their indices).  The new elements
         /// will appear in this list in the order that they are returned by the
         /// specified collection's iterator.  The behavior of this operation is
         /// undefined if the specified collection is modified while the
         /// operation is in progress.  (Note that this will occur if the specified
         /// collection is this list, and it's nonempty.)
-        /// </summary>
-        /// <param name="index">index at which to insert the first element from the specified collection</param>
-        /// <param name="items">collection containing elements to be added to this list</param>
-        /// <typeparam name="TItem"></typeparam>
-        /// <returns><tt>true</tt> if this list changed as a result of the call</returns>
+        /// </remarks>
         Task<bool> InsertRangeAsync<TItem>(int index, ICollection<TItem> items) where TItem : T;
 
         /// <summary>
-        /// Replaces the element at the specified position in this list with the
-        /// specified element.
+        /// Replaces the element at the specified position in this list with the specified element.
         /// </summary>
         /// <param name="index">index index of the element to replace</param>
         /// <param name="item">element to be stored at the specified position</param>
-        /// <returns></returns>
+        /// <returns>The element previously at the specified position</returns>
         Task<T> SetAsync(int index, T item);
```

Comment:
If we return a value should this be `GetAndSetAsync` just to be consistent?

Reply:
ok changing the RemoveAt too